src/main/java/io/github/addoncommunity/galactifun/base/milkyway/solarsystem/aliens/Leech.java

```diff
@@ -0,0 +1,88 @@
+package io.github.addoncommunity.galactifun.base.milkyway.solarsystem.aliens;
+
+import io.github.addoncommunity.galactifun.api.universe.world.Alien;
+import io.github.mooy1.infinitylib.core.PluginUtils;
+import io.github.mooy1.infinitylib.items.PersistentStackArray;
+import org.bukkit.NamespacedKey;
+import org.bukkit.attribute.Attribute;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.entity.EntityDeathEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.PlayerInventory;
+import org.bukkit.persistence.PersistentDataContainer;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Class for the leech, an alien of Titan. They can steal you items when attacking
+ *
+ * @author Seggan
+ */
+public final class Leech extends Alien {
+
+    private static final NamespacedKey EATEN = PluginUtils.getKey("eaten");
+    
+    public Leech() {
+        super("LEECH", "&eLeech", EntityType.SILVERFISH, 10);
+    }
+
+    @Override
+    protected void onAttack(@Nonnull EntityDamageByEntityEvent e) {
+        
+        // get random item
+        PlayerInventory inv = ((Player) e.getEntity()).getInventory();
+        
+        int slot = ThreadLocalRandom.current().nextInt(inv.getSize());
+        
+        ItemStack item = inv.getItem(slot);
+        
+        if (item == null) {
+            return;
+        }
+        
+        // eat it
+        PersistentDataContainer container = e.getEntity().getPersistentDataContainer();
+        
+        ItemStack[] eatenItems = container.get(EATEN, PersistentStackArray.instance());
+        if (eatenItems != null) {
+            // add on to the array
+            ItemStack[] arr = new ItemStack[eatenItems.length + 1];
+            System.arraycopy(eatenItems, 0, arr, 0, eatenItems.length);
+            arr[eatenItems.length] = item;
+            container.set(EATEN, PersistentStackArray.instance(), arr);
+        } else {
+            container.set(EATEN, PersistentStackArray.instance(), new ItemStack[] {item});
+        }
+        
+        inv.setItem(slot, null);
+        
+        // heal
+        LivingEntity attacker = (LivingEntity) e.getDamager();
+        attacker.setHealth(Math.min(
+                Objects.requireNonNull(attacker.getAttribute(Attribute.GENERIC_MAX_HEALTH)).getBaseValue(),
+                attacker.getHealth() + 2)
```

Comment:
```suggestion
        attacker.setHealth(Math.min(
                this.maxHealth,
                attacker.getHealth() + 2)
```
since you made the health var `protected`, might as well do this